src/traces/scattergl/convert.js

```diff
@@ -264,6 +290,15 @@ proto.update = function(options) {
         this.updateFast(options);
     }
 
+    // sort objects so that order is preserve on updates:
+    // - lines
+    // - errorX
+    // - errorY
+    // - markers
+    this.scene.glplot.objects.sort(function(a, b) {
```

Comment:
This part isn't quite right. We'll have to consider trace order too, fix coming on Monday.

The same technique might be extended to 3D to fix https://github.com/plotly/plotly.js/issues/1267